flooder/src/main.rs

```diff
@@ -1,100 +1,258 @@
+                        )?
+                        .submit(status)
+                        .await?;
+                    }
+                    anyhow::Ok(())
                 }
             })
-        })
-        .collect();
+            .collect::<Vec<_>>();
 
-    join_all(handles).await;
+        for result in join_all(tasks).await {
+            if let Err(e) = result {
+                info!("Error submitting transaction: {e:?}");
+                return Err(e);
+            }
+        }
+
+        let duration = Instant::now().saturating_duration_since(start);
+        if duration.as_secs() >= schedule.interval_duration * schedule.intervals {
+            break;
+        }
+        let left_duration =
+            Duration::from_secs((i + 1) * schedule.interval_duration).saturating_sub(duration);
+        sleep(left_duration).await;
+    }
+
+    info!("Flooding time has passed, left {overdue_transactions} overdue txns because of the pool limit.");
+    if return_balance_at_the_end {
+        for (conn_id, conn) in connections.iter().enumerate() {
+            conn.sign_with_params(
+                transfer_all(dest.clone(), true),
+                Default::default(),
+                nonces[conn_id],
+            )?
+            .submit(status)
+            .await?;
+        }
+        debug!("Returned balance back to main account");
+    }
 
     Ok(())
 }
 
 async fn initialize_n_accounts<F: Fn(u32) -> String>(
-    connection: SignedConnection,
+    connection: &SignedConnection,
     n: u32,
     node: F,
-    account_balance: Balance,
+    amount: Balance,
     skip: bool,
-) -> Vec<SignedConnection> {
+) -> anyhow::Result<Vec<SignedConnection>> {
+    info!(
+        "Initializing accounts, estimated total fee per account: {}TZERO",
+        amount as f32 / TOKEN as f32
+    );
     let mut connections = vec![];
     for i in 0..n {
         let seed = i.to_string();
-        let signer = KeyPair::new(raw_keypair_from_string(&("//".to_string() + &seed)));
+        let signer = KeyPair::new(raw_keypair_from_string(
+            &(ACCOUNTS_SEED_PREFIX.to_string() + &seed),
+        ));
         connections.push(SignedConnection::new(&node(i), signer).await);
     }
 
     if skip {
-        return connections;
+        return Ok(connections);
     }
-    for conn in connections.iter() {
+
+    let nonce = connection.account_nonce(connection.account_id()).await?;
+    for (i, conn) in connections.iter().enumerate() {
         connection
-            .transfer_keep_alive(
-                conn.account_id().clone(),
-                account_balance,
-                TxStatus::Submitted,
-            )
-            .await
-            .unwrap();
+            .sign_with_params(
+                transfer_keep_alive(conn.account_id().clone(), amount),
+                Default::default(),
+                nonce + i as Nonce,
+            )?
+            .submit(TxStatus::Submitted)
+            .await?;
     }
 
     connection
```

Comment:
What is for this transfer?

Reply:
It was a trick (a transfer to itself indeed), so that we can wait on finalization of this transaction instead of awaiting on all the previous ones :P . Removed and added waiting for last transaction finalization in the loop.